tools/flowc/incremental/fi_type_unify.flow

```diff
@@ -187,13 +188,13 @@ fiUnifyTypes(type1 : FiType, type2: FiType) -> Pair<FiType, FiType> {
 	}
 }
 
-fiMakeCast(e: FiExp, from: FiType, to: FiType) -> FiExp {
+fiMakeCast(e: FiExp, from: FiType, to: FiType, subflags: FiSubtypeFlags, names: FiGlobalNames) -> FiExp {
 	if (fiEqualTypes(from, to) || to == FiTypeVoid()) e else {
 		switch (fiMatchTyparsSym(from, to, Pair(makeTree(), makeTree()))) {
 			Some(p): {
 				from1 = fiApplyTypeSubstitution(p.first, from);
 				to1 = fiApplyTypeSubstitution(p.second, to);
-				if (fiEqualTypes(from1, to1)) e else FiCast(e, from1, to1, to1, e.start);
```

Comment:
Will this not introduce excessive casts that are not required?

Reply:
Sometimes such casts are needed. Otherwise I didn't introduce this operation. The main idea: make _any_ casts explicit, and after that a backend decides, whether in a particular case it makes sense or not (in this particular backend).